### PR TITLE
feat(codescoop): osman integration

### PR DESCRIPTION
**Pull request by sweetca**
Codescoop integration:

1. Maven profile to enable Codescoop inside sw360
2. Thrift Service for CodescoopHandler servlet
3. OSMAN dependency for API access
4. Purl integration for matching components between systems

### CODE REVIEW

**Comment by maxhbr** [comment]
@sweetca please add the sign-off footer since (see: https://accounts.eclipse.org/legal/eca/validation/1282):
> Alex Skorohod (alexey.skorohod@****.com) did not include the "Signed-off-by footer" which is required for all commits

**Comment by mcjaeger** [comment]
@sweetca please care for merge conflicts

**Comment by mcjaeger** [comment]
@sweetca could you pls have a look at the source code questions?

File: frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java [hunk 1]
```diff
@@ -448,10 +458,60 @@ private void serveReleaseSearchResults(ResourceRequest request, ResourceResponse
...
+    }
+
     //! VIEW and helpers
     @Override
     public void doView(RenderRequest request, RenderResponse response) throws IOException, PortletException {
         String pageName = request.getParameter(PAGENAME);
+
+        try {
+            CodescoopService.Iface codescoopClient = thriftClients.makeCodescoopClient();
+            request.setAttribute("codescoopActive", codescoopClient.isEnabled());
```
[the remark this answers is not in the document]
Comment (sweetca) [reply frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/components/ComponentPortlet.java]: @smrutis1 such flow will not work - codescoop connection may fall down or re-connect during sw360 runtime

**Comment by smrutis1** [comment]
Code looks good now.

**Pull request merged**